app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -386,10 +386,13 @@ def remove_middleware_datasource(ems_ref)
     end
 
     def emit_middleware_notification(type, op_name, op_arg, ems_ref)
+      mw_deployment = MiddlewareDeployment.find_by_ems_ref(ems_ref)
+      mw_server = mw_deployment.nil? ? MiddlewareServer.find_by_ems_ref(ems_ref) :
```

Comment:
Yes, it's an identifier string that can be the id of either a MiddlewareDeployment or a MiddlewareServer (and in the future, possibly, of a MiddlewareDatasource/Domain/Messaging, etc..). I wanted a different approach as well but this is the best my RoR knowledge could get. Do you see other options ?